cmd/global-query/pkg/distributed/querier.go

```diff
@@ -28,6 +29,14 @@ type Querier interface {
 	CreateQueryWorkload(ctx context.Context, host string, args *query.Args) (*QueryWorkload, error)
 }
 
+// QuerierAnyable extends a "common" Querier with the support to retrieve a list of all hosts / targets
+// available to the Querier
+type QuerierAnyable interface {
+
+	// AllHosts returns a list of all hosts / targets available to the Querier
+	AllHosts() (hosts.Hosts, error)
```

Comment:
I'd go for `ResolveAny()` as it is more in line with the constant for `"any"` you define later on.